packages/jest-config/src/__tests__/normalize.test.js

```diff
@@ -964,7 +995,7 @@ describe('preset without setupFiles', () => {
 
   beforeAll(() => {
     jest.mock(
-      '/node_modules/react-native/jest-preset.json',
+      '/node_modules/react-foo/jest-preset.json',
```

Comment:
It seems to be some kind of bug here. When it mocks the same file name as in the other `describe` block, it doesn't reload the config but uses what was previously set. I updated this name to ensure that `setupFiles` is configured properly, i.e. `babel-jest` is enabled which triggers prepend of `regenerator-runtime`.

Reply:
We could switch to `jest.doMock` here to avoid this, since it's already placed before the require call and we don't need to hoist it.

Reply:
Let me try it. This was my guess that `jest.mock` calls get hoisted and this flow doesn't work as expected.